tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -586,23 +587,17 @@ bool IsHLSLSubobjectType(clang::QualType type) {
   return GetHLSLSubobjectKind(type, kind, hgType);
 }
 
-bool IsUserDefinedRecordType(clang::QualType type) {
-  if (const auto *rt = type->getAs<RecordType>()) {
-    // HLSL specific types
-    if (hlsl::IsHLSLResourceType(type) || hlsl::IsHLSLVecMatType(type) ||
-        isa<ExtVectorType>(type.getTypePtr()) || type->isBuiltinType() ||
-        type->isArrayType()) {
-      return false;
-    }
-
-    // SubpassInput or SubpassInputMS type
-    if (rt->getDecl()->getName() == "SubpassInput" ||
-        rt->getDecl()->getName() == "SubpassInputMS") {
+bool IsUserDefinedRecordType(clang::QualType QT) {
```

Comment:
This changes the behavior of the function, at least for `BuiltInTriangleIntersectionAttributes`, which would have returned true before.

Reply:
Upon review, I think this change is ok for this function based on the name, and based on a survey of uses.